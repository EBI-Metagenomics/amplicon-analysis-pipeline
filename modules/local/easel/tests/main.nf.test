nextflow_process {

    name "Test Process EASEL"
    script "../main.nf"
    process "EASEL"
    tag "modules"
    tag "easel"

    test("Should take a fasta file and a cmsearch tblout.deoverlapped file and generate two coordinate files for subunit extraction") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('${projectDir}/modules/local/easel/tests/data/test.fasta.gz', checkIfExists: true),
                    file('${projectDir}/modules/local/easel/tests/data/test.tblout.deoverlapped', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.easel_coords).match("easel_coords") },
                { assert snapshot(process.out.matched_seqs_with_coords).match("matched_seqs_with_coords") }
            )

        }

    }

}
