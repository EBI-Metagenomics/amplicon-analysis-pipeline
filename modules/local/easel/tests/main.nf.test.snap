{
    "easel_coords": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.easel_extracted.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                ]
            ]
        ],
        "timestamp": "2023-11-02T13:03:08.601689336"
    },
    "matched_seqs_with_coords": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.matched_seqs_with_coords:md5,401222d263116bb145c1a67e6acc4709"
                ]
            ]
        ],
        "timestamp": "2023-11-02T13:03:08.640389187"
    }
}